astropy/io/fits/header.py

```diff
@@ -1694,11 +1694,11 @@ def _keyword_from_index(self, idx):
         """
 
         if idx < 0:
-            idx += len(self._cards) - 1
+            idx += len(self._cards)
 
         keyword = self._cards[idx].keyword
         keyword = Card.normalize_keyword(keyword)
-        repeat = self._keyword_indices[keyword].index(idx)
+        repeat = len(self._keyword_indices[keyword]) - 1
```

Comment:
Exactly, I made the change based on docstring and what @rsiverd said in the issue. Specifically 
```
...
or most keywords the repeat will always be zero, but it may be greater than zero for keywords that are duplicated (especially commentary keywords).
```
What do you suggest I do? I am seeing the compressed module. Was unaware of that.